src/components/billboard.js

```diff
@@ -0,0 +1,66 @@
+const isMobileVR = AFRAME.utils.device.isMobileVR();
+
+// Billboard component that only updates visible objects and only those in the camera view on mobile VR.
+AFRAME.registerComponent("billboard", {
+  init: function() {
+    this.target = new THREE.Vector3();
+    this._updateBillboard = this._updateBillboard.bind(this);
+    this._updateIsInView = this._updateIsInView.bind(this);
+
+    if (isMobileVR) {
+      this.el.sceneEl.systems["frame-scheduler"].schedule(this._updateIsInView, "billboards");
+    }
+  },
+  remove() {
+    if (isMobileVR) {
+      this.el.sceneEl.systems["frame-scheduler"].unschedule(this._updateIsInView, "billboards");
+    }
+  },
+
+  tick() {
+    if (this.isInView || !isMobileVR) {
+      this._updateBillboard();
+    }
+  },
+
+  _updateIsInView: (function() {
+    const frustum = new THREE.Frustum();
+    const frustumMatrix = new THREE.Matrix4();
+    const cameraWorld = new THREE.Vector3();
+    const isInViewOfCamera = (screenCamera, pos) => {
+      frustumMatrix.multiplyMatrices(screenCamera.projectionMatrix, screenCamera.matrixWorldInverse);
+      frustum.setFromMatrix(frustumMatrix);
+      return frustum.containsPoint(pos);
+    };
```

Comment:
It's kind of frustrating that we aren't reusing the work that the renderer is doing to determine whether things are frustum-culled. I can see that it's not so easy, though, because if we reuse the renderer's `_frustum` property we'll be one frame behind. I guess in the ideal world we would need to rework the order of operations to first find the view frustum for the upcoming frame, then run the application tick, respecting that frustum, and then run the renderer, respecting that frustum.

Reply:
for the purposes of this kind of optimization being a frame behind wouldn't be a big deal -- but i think a three.js patch to do this would be iffy because it'd mean at least one more branch and a bunch of memory writes each frame per object

Reply:
You wouldn't need a three.js patch to do it. You would just use the `_frustum` property that already gets set every frame on the renderer object.

I have yet to be convinced that being a frame behind on things is not a big deal. It seems like common sense to me that maximally crisp-feeling responsiveness, state transitions, and behavior is important in making games and other software "feel good", and I have heard plenty of design stories about smart game developers working hard to cut frame delays on things, so I see no reason to doubt my common sense. I think we just wish it wasn't a big deal because it's hard.

Reply:
I was referring to billboarding specifically, and our use of it at that. A frame delay of the nametags being rotated to face you would be imperceptible.